### ISSUE TITLE
Memory leak under heavy UDP traffic via socks5 

### ISSUE THREAD

**Issue by comzyh** [Dreamacro/clash#436]
I wrote a [stress test](https://gist.github.com/comzyh/18c02a2a043ae59063d3220c5ea2d628) and test on clash latest dev(version 77b5d63d296f72a3e2c6f367b92680d73c9cb739). I guess I encountered a memory leak issue.

Reproduce
---

1. start clash and ensure clash accept socks5 traffic
> $ bin/clash-linux-amd64 -v
> Clash v0.10.2-187-g77b5d63 linux amd64 2019年 12月 18日 星期三 09:47:02 UTC
> $bin/clash-linux-amd64

![image](https://user-images.githubusercontent.com/1068203/71075445-0a177100-21bf-11ea-9614-c820915e2ff1.png)

2. download clash_stress.go 

> wget https://gist.githubusercontent.com/comzyh/18c02a2a043ae59063d3220c5ea2d628/raw/488027435e7836625b7a75908433ec3690d42136/clash_stress.go

3. run clash_stress test

**WARNING**: You need to stop the test as soon as possible in case the memory is full
```shell
go run clash_stress.go -a 127.0.0.1 -p <clash_socks5_port> -socks5 127.0.0.1:12345
```
Clast may take  GBs of memory
![image](https://user-images.githubusercontent.com/1068203/71075641-611d4600-21bf-11ea-893a-e7fcfb1baaf5.png)

4. Stop the test and check the memory usage again

![image](https://user-images.githubusercontent.com/1068203/71075864-bd806580-21bf-11ea-8465-9fa6ea89e5ba.png)

The memory usage will not go down, I think this may be a **memory leak**

Possible fix
---

There is only one goroutine that handling the UDP packets

https://github.com/Dreamacro/clash/blob/e5284cf647717a8087a185d88d15a01096274bc2/tunnel/tunnel.go#L100-L106

Maybe clash can start more goroutine to handle the UDP packets?

**Comment by comzyh** [comment]
related #268 #265

**Comment by BirkhoffLee** [comment]
Fixed in #441
